### ISSUE TITLE
Support for a real Checkmark in the Checkbox?

### ISSUE THREAD

**Issue by volcoma** [ocornut/imgui#593]
I was wondering if an actual checkmark will be added to the checkbox? Is there a reason it was not added in the first place? Its pretty easy to do so. Here is an example code that replaces the exsisting part in the code

```
if (*v)
    {
        const float check_sz = ImMin(check_bb.GetWidth(), check_bb.GetHeight());
        const float pad = ImMax(1.0f, (float)(int)(check_sz / 6.0f));
        //window->DrawList->AddRectFilled(check_bb.Min+ImVec2(pad,pad), check_bb.Max-ImVec2(pad,pad), GetColorU32(ImGuiCol_CheckMark), style.FrameRounding);
        const ImRect check_mark_bb(check_bb.Min + ImVec2(pad, pad), check_bb.Max - ImVec2(pad, pad));
        const ImVec2 poitns[3] = { ImVec2(check_mark_bb.GetCenter().x - check_mark_bb.GetWidth()/2, check_mark_bb.GetCenter().y), ImVec2(check_mark_bb.GetCenter().x, check_mark_bb.GetCenter().y + check_mark_bb.GetHeight() / 2), ImVec2(check_mark_bb.GetTR()) };
        window->DrawList->AddPolyline(poitns, 3, GetColorU32(ImGuiCol_CheckMark), false, 3.0f, true);
    }
```

And here is how it looks:
![checkmark](https://cloud.githubusercontent.com/assets/1499411/14582117/f174c89a-0404-11e6-8c58-d8cb9ec40362.png)

**Comment by ratchetfreak** [comment]
FYI if you use 3 "\`" you can keep the formatting:

```C
if (*v) { 
    const float check_sz = ImMin(check_bb.GetWidth(), check_bb.GetHeight()); 
    const float pad = ImMax(1.0f, (float)(int)(check_sz / 6.0f)); 
    //window->DrawList->AddRectFilled(check_bb.Min+ImVec2(pad,pad), check_bb.Max-ImVec2(pad,pad), GetColorU32(ImGuiCol_CheckMark), style.FrameRounding); 
    const ImRect check_mark_bb(check_bb.Min + ImVec2(pad, pad), check_bb.Max - ImVec2(pad, pad)); 
    const ImVec2 poitns[3] = { ImVec2(check_mark_bb.GetCenter().x - check_mark_bb.GetWidth()/2, check_mark_bb.GetCenter().y), ImVec2(check_mark_bb.GetCenter().x, check_mark_bb.GetCenter().y + check_mark_bb.GetHeight() / 2), ImVec2(check_mark_bb.GetTR()) }; 
    window->DrawList->AddPolyline(poitns, 3, GetColorU32(ImGuiCol_CheckMark), false, 3.0f, true);
}
```

You can even set it to add higlighting by adding the language

**Comment by ocornut** [comment]
I didn't have thick polyline with proper connections back then.
The other reason is that I didn't want 1-pixel thin shapes that looked like fonts but when wouldn't match variety of fonts. So I went for a square.

There's still a bug with thick polyline normal merging in the non-antialiased path which would need to be fixed before we can use that (see it here: https://github.com/ocornut/imgui/pull/288#issuecomment-126555921 )

Ideally we'd want a) a symbol library b) those would rasterized in the font atlas so they would cost 2 triangles, same principle would apply for rounded corners which is the primary reason for rasterizing stuff into the atlas. Of course we don't need to do any of that to add that check symbol, just need to fix the non-AA path to match the normal merging of the AA-path.

**Comment by ocornut** [comment]
You can see the bug easily in the Custom Rendering demo if you disable anti-aliasing:
![capture](https://cloud.githubusercontent.com/assets/8225057/14585869/9d96433a-0485-11e6-8898-fd5f0798ed48.PNG)

**Comment by dougbinks** [comment]
FYI if you're prepared to load a font like Font Awesome you can use the CheckBoxTick from this gist: https://gist.github.com/dougbinks/8089b4bbaccaaf6fa204236978d165a9

**Comment by franciscod** [comment]
This is already merged on bf778ebb -- unless I'm missing something of course :)

**Comment by ocornut** [comment]
Correct, it's now in imgui (with the exact issue I described if anti-aliasing is disabled, need to fix the non-AA path). I'll close this issue as it is in the TODO to fix them and notably RenderCheckmark().

Down the line, the imgui_internal.h functions such as RenderCheckMark(), RenderTriangle(dir), RenderBullet() should be refactored as e.g. RenderSymbol(enum). Would like to see how we could leverage the recent CustomRect API of ImFontAtlas to have an already-rasterized path for them too.
